sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -133,6 +144,32 @@ object JDBCRDD extends Logging {
     })
   }
 
+  def compileAggregates(
+      aggregates: Seq[AggregateFunc],
+      dialect: JdbcDialect): (Array[String]) = {
+    def quote(colName: String): String = dialect.quoteIdentifier(colName)
+
+    val aggBuilder = ArrayBuilder.make[String]
+    aggregates.map {
```

Comment:
Sorry I don't get it. Why not just `aggregates.map`? Why do we need an ArrayBuilder?